src/utils/hooks.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useRef } from 'react';
+
+// Hook
+function usePrevious<T>(value: T): T {
```

Comment:
What will this be used for? like in login | register ?

I propose that if it's not related to the current change, this should be a separate pull request :3 so this request will only be for any changes regarding the register page :)

Reply:
Since I already did it, I'd prefer to not move it to a different PR for no reason :3

Reply:
@allie-rae the reason is to conform to how version control should work should there be any changes in the future, we can follow the commit messages and revert and for us to keep track ourselves for when the changes are being incorporated :) I understand it is more convenient for us to do it this way now, but please do keep this in mind for future PR